discord/voice_client.py

```diff
@@ -291,8 +306,188 @@ def is_connected(self):
         """bool: Indicates if the voice client is connected to voice."""
         return self._connected.is_set()
 
+    def event(self, coro):
+        """A decorator that registers an event to listen to.
+
+        The events must be a |corourl|_, if not, :exc:`ClientException` is raised.
+
+        The currently supported events are:
+
+        * `on_receive_raw(sequence, timestamp, user_id, data)`
+        * `on_receive_pcm(sequence, timestamp, user_id, data)`
+        * `on_speaking_start(user_id)`
+        * `on_speaking_stop(user_id)`
+
+        Example
+        ---------
+        ::
+            @voice_client.event
+            @asyncio.coroutine
+            def on_speaking_start(user_id):
+                print('The user with the ID {} has started talking'.format(user_id))
+        """
+
+        if not asyncio.iscoroutinefunction(coro):
+            raise ClientException('event registered must be a coroutine function')
+
+        if coro.__name__ not in self._events:
+            raise ClientException('no event found named {}'.format(coro.__name__))
+
+        self._events[coro.__name__] = coro
+        log.info('%s has successfully been registered as an event', coro.__name__)
+        return coro
+
     # audio related
 
+    @asyncio.coroutine
+    def get_voice_data(self, user, timeout=400):
+        """|coro|
+
+        Receive voice data from a user untill they are silent for
+        `timeout` ms.
+
+        Parameters
+        -----------
+        user: :class:`abc.User`
+            The user to listen for audio from.
+        timeout: Optional[int]
+            The time in ms of silence after which to stop listening.
+
+        Returns
+        -----------
+        The received voice data for the user passed in `bytes` form.
+
+        The data is stereo 48kHz audio in raw PCM. You will have to add your
+        own headers if you want to save it as a `.wav` file.
+        """
+        global buffer, last_packet
+        data = b''
```

Comment:
This should probably be a `bytearray` instead of `bytes` since that is actually mutable.